src/app/cli/src/commands/verify_command.rs

```diff
@@ -76,24 +78,56 @@ impl VerifyCommand {
         options: VerificationOptions,
         listener: Option<Arc<VerificationMultiProgress>>,
     ) -> GenericVerificationResult {
-        let dataset_handle = self
-            .dataset_repo
-            .resolve_dataset_ref(self.refs.first().unwrap())
-            .await?;
-
-        let listener = listener.and_then(|l| l.begin_verify(&dataset_handle));
+        let dataset_ref_pattern = self.refs.first().unwrap();
+        match dataset_ref_pattern {
+            DatasetRefPattern::Ref(dataset_ref) => {
+                let dataset_handle = self.dataset_repo.resolve_dataset_ref(dataset_ref).await?;
+
+                let listener = listener.and_then(|l| l.begin_verify(&dataset_handle));
+
+                let res = self
+                    .verification_svc
+                    .verify(
+                        &dataset_handle.as_local_ref(),
+                        (None, None),
+                        options,
+                        listener,
+                    )
+                    .await;
+
+                Ok(vec![Ok(VerificationMultiResult {
+                    dataset_handle,
+                    verification_result: res,
+                })])
```

Comment:
Let's move this into a method too. It's usually much more readable when all parts of a function work at the same level of abstraction.